Algorithm/QCAlgorithm.History.cs

```diff
@@ -406,6 +400,12 @@ public IEnumerable<TradeBar> History(Symbol symbol, DateTime start, DateTime end
                 Error("Calling History<TradeBar> method on a Forex or CFD security will return an empty result. Please use the generic version with QuoteBar type parameter.");
             }
 
+            if (resolution.HasValue && resolution == Resolution.Tick)
```

Comment:
If the `resolution` value is null, will this check be bypassed if the only subscription we have for a given `Symbol` is in `Resolution.Tick`? 

We select the highest resolution available in the added Security if it exists in the following method, which is called from `CreateDateRangeHistoryRequests`:
https://github.com/QuantConnect/Lean/blob/a3c5fd98712be4e8416f08bfb548957e9aabcb4a/Algorithm/QCAlgorithm.History.cs#L707

Reply:
Nice catch! Addressed 👍